src/OrchardCore.Modules/OrchardCore.Workflows/Controllers/WorkflowTypeController.cs

```diff
@@ -122,23 +126,21 @@ public async Task<IActionResult> Index(WorkflowTypeIndexOptions options, PagerPa
                 .Take(pager.PageSize)
                 .ListAsync();
 
-            var connection = await _session.CreateConnectionAsync();
```

Comment:
Can't see the dispose here

Reply:
yes. As explain in the meeting today, `IDbQueryExecutor` will prevent a bad practice like this from happening. There are multiple useful extension added on `IDbQueryExecutor` that would make executing query much easier.

Reply:
@sebastienros @MikeAlhayek 

Here this is not the same, we call `CreateConnectionAsync()` on the `ISession` which is a way to get (or create if not yet done) the `_connection` held by `ISession`.

So because we ensure that `ISession` is disposed at the end of a shell scope, this `_connection` will be closed and disposed when the `ISession` will be disposed.